include/task/state_machine_types.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file state_machine_types.hpp
+ *
+ * @brief Useful types for the state machine and other tasks
+ *
+ * @date 06/2024
+ */
+
+#ifndef __STATE_MACHINE_TYPES_HPP__
+#define __STATE_MACHINE_TYPES_HPP__
+
+#include <Arduino.h>
+
+namespace rfidoor::task {
+
+#define NUM_STATES 7
+#define NUM_EVENTS 12
+
+/**
+ * @brief Enum corresponding to the possible states of the state machine
+ */
+enum state_t : int8_t {
+  TRANCADA_IDLE,
+  DIGITANDO_SENHA,
+  DESTRANCADA_FECHADA,
+  ABERTA,
+  REGISTRO
+};
+
+/**
+ * @brief Enum corresponding to the possible events of the state machine
+ */
+enum event_t : int8_t {
+  NENHUM_EVENTO = -1,
+  SINAL_INVALIDO,
+  SINAL_VALIDO,
+  SINAL_CADASTRADO,
+  BOTAO,
+  TIMEOUT,
+  SENHA_INVALIDA,
+  SENHA_VALIDA,
+  SENHA_CADASTRADA,
+  TECLA,
+  FECHAR,
+  ABRIR,
+  SEGURAR_ASTERISCO
+};
+
+/**
+ * @brief Enum corresponding to the possible actions of the state machine
+ */
+enum action_t : int8_t {
+  NENHUMA_ACAO = -1,
+  A01,
+  A02,
+  A03,
+  A04,
+  A05,
+  A06,
+  A07,
+  A08,
+  A09,
+  A10,
+  A11,
+  A12,
+  A13,
+  A14
+};
+
+/**
+ * @brief Enum corresponding to the possible states of the keyboard and rfid
+ * sensor
+ *
+ * @note rfid and keyboard can read password or id, and register password or id
+ */
+enum input_device_state_t : int8_t { READING, REGISTERING };
+
```

Comment:
Não acho que seja necessário essa divisão. Deixar só os estados padrões para todos.

Reply:
Po eu acho bacana essa divisão justamente porque são duas filas diferentes, tipo assim, não faz sentido usar os estados da fila EVENTO pra fila LER REGISTRAR

Reply:
Tipo assim, eu acho que esse esquema de Tasks

![grafik](https://github.com/pedroazeredo04/RFIDoor/assets/89558427/926b8eea-4b6a-4720-8ae3-954e9cf0816a)

É mais elegante e faz mais sentido que esse esquema de Tasks

![grafik](https://github.com/pedroazeredo04/RFIDoor/assets/89558427/b89e2de4-5517-4393-b3be-6690eae32262)

Reply:
Mas os Estados não são de fila de nada, a fila eventos manda eventos, a task que processa e muda os estados. Tendo uma fila pra todos os estados acredito que facilita a utilização na senha também que é diferente, e caso a gente for usar na main para displays padrões etc, por exemplo, sempre que mudar pra `TRANCADA_IDLE` display digitar senha, assim fica mais simples de fazer displays temporários também!

Reply:
Eu ainda não gosto de tasks dependerem da máquna de estados, eu acharia melhor se eu pudesse mudar a máquina de estados sem mudar todas as tasks. Tendo dito isso, vou mudar e é isso aí mesmo